roomserver/storage/shared/storage.go

```diff
@@ -823,13 +823,38 @@ func (d *Database) handleRedactions(
 		return nil, "", nil
 	}
 
+	// Get the power level from the database, so we can verify the user is allowed to redact the event
+	powerLevels, err := d.GetStateEvent(ctx, event.RoomID(), gomatrixserverlib.MRoomPowerLevels, "")
+	if err != nil {
+		return nil, "", fmt.Errorf("d.GetStateEvent: %w", err)
+	}
+	pl, err := powerLevels.PowerLevels()
+	if err != nil {
+		return nil, "", fmt.Errorf("unable to get powerlevels for room: %w", err)
+	}
+	redactPL := pl.Redact
+	redactUser := pl.Users[redactionEvent.Sender()]
```

Comment:
Please use `pl.UserLevel()` rather than accessing the map directly, so that it does the right thing if the defaults have been elevated or the redaction level has been reduced.